build_tools/github_actions/runner/README.md

```diff
@@ -213,12 +213,24 @@ changes you make need to be forward and backward compatible with changes to
 anything that is picked up directly from tip of tree (such as workflow files).
 These should be changed in separate PRs.
 
-To deploy to prod, create new prod templates. Then use the update script to
-canary to a single instance:
+To deploy to prod, create new prod templates. Then canary the new template for
+one instance in each group.
+
+Note: The a100 groups are special. We only run one instance in each group and
+have one of every type in every region, so canarying within a single instance
+group doesn't really make any sense. Also, we use the `balanced` target
+distribution shape, which theoretically means that the group manager will avoid
+zones with no available capacity (which happens a lot). This distribution shape
+is for some reason incompatible with having multiple templates. So in the
+canarying below, we treat these differently.
 
 ```shell
 build_tools/github_actions/runner/gcp/update_instance_groups.py canary \
-  --env=prod --region=all --group=all --type=all --version="${VERSION}"
+  --env=prod --region='us-\w+' --group=all --type='[^a]\w+' \
```

Comment:
Why do we use `--region='us-\w+'` instead of `--region=all` here?

Reply:
Because searching across all regions is slow and we have everything in the US. I could drop that if it seems to confusing here though

Reply:
I see. It might be a surprise if we have MIG in other regions. How much time did it take to scan all regions? If the diff is significant then it's definitely worth limiting the regions.

Reply:
Takes about twice as long/an extra 11s (for the whole thing): 22s vs 11s

```shell
$ time build_tools/github_actions/runner/gcp/update_instance_groups.py direct-update \
    --env=prod \
    --region='us-\w+' \
    --group=all \
    --type=all \
    --version=2f4d5acd90-6ts \
    --dry-run \
    --skip-confirmation
[...]
real    0m11.467s
user    0m3.370s
sys     0m0.520s

$ time build_tools/github_actions/runner/gcp/update_instance_groups.py direct-update \
    --env=prod \
    --region=all \
    --group=all \
    --type=all \
    --version=2f4d5acd90-6ts \
    --dry-run \
    --skip-confirmation
[...]
real    0m22.698s
user    0m4.593s
sys     0m0.863s
```

This includes printing out all the commands it's going to run, and such. If we early exit after finding the migs it's 18-19s vs 7-8s (so basically the rest of the script takes 3s)

Reply:
Oh, I see. That's a lot. I have no problem with the current change.

Reply:
I added a note in the docs. Thanks for asking the question.

Reply:
Oh, actually I updated the docs in my other PR. Hang on...